dotnet/src/SemanticKernel.Abstractions/SemanticFunctions/PromptConfig.cs

```diff
@@ -0,0 +1,278 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.TemplateEngine;
+using Microsoft.SemanticKernel.Text;
+using static Microsoft.SemanticKernel.SemanticFunctions.PromptTemplateConfig;
+
+namespace Microsoft.SemanticKernel.SemanticFunctions;
+
+/// <summary>
+/// Prompt configuration.
+/// </summary>
+public class PromptConfig
+{
+    /// <summary>
+    /// Input parameter for semantic functions.
+    /// </summary>
+    public class InputParameter
```

Comment:
As an example, regarding naming, here we have `prompt` configuration, but input parameter for `semantic functions`. For users it may be confusing and not clear whether it's the same thing or not. I would probably try to align the naming in scope of this refactoring/redesign.